docs-gen/content/docs/tools/cli.md

```diff
@@ -4,8 +4,584 @@ weight: 1
+  side: SideEnum
+}
+
+enum RowEnum {
+  ROW1
+  ROW2
+}
+
+enum SideEnum {
+  DRIVERSIDE
+  PASSENGERSIDE
+}
+```
+
+The Protobuf exporter produces:
+
+```protobuf
+syntax = "proto3";
+
+import "google/protobuf/descriptor.proto";
+import "buf/validate/validate.proto";
+
+extend google.protobuf.MessageOptions {
+  string source = 50001;
+}
+
+message RowEnum {
+  option (source) = "RowEnum";
+
+  enum Enum {
+    ROWENUM_UNSPECIFIED = 0;
+    ROW1 = 1;
+    ROW2 = 2;
+  }
+}
+
+message SideEnum {
+  option (source) = "SideEnum";
+
+  enum Enum {
+    SIDEENUM_UNSPECIFIED = 0;
+    DRIVERSIDE = 1;
+    PASSENGERSIDE = 2;
+  }
+}
+
+message DoorPosition {
+  option (source) = "DoorPosition";
+
+  RowEnum.Enum row = 1;
+  SideEnum.Enum side = 2;
+}
+
+message Cabin {
+  option (source) = "Cabin";
+
+  repeated Door doors = 1;
+  float temperature = 2;
+}
+
+message Door {
+  option (source) = "Door";
+
+  bool isLocked = 1;
+  DoorPosition instanceTag = 2;
+}
+```
+
+#### Selection Query (Required)
+
+The protobuf exporter requires a selection query to determine which types and fields to export:
+
+```bash
+s2dm export protobuf --schema schema.graphql --selection-query query.graphql --output cabin.proto
+```
+
+Given a query file `query.graphql` that matches the schema above:
+
+```graphql
+query Selection {
+  cabin {
+    doors {
+      isLocked
+      instanceTag {
+        row
+        side
+      }
+    }
+    temperature
+  }
+}
+```
+
+The exporter will include only the selected types and fields from the schema.
+
+#### Root Type Filtering
+
+Use the `--root-type` flag in combination with the selection query to further filter the export:
+
+```bash
+s2dm export protobuf --schema schema.graphql --selection-query query.graphql --output cabin.proto --root-type Cabin
+```
+
+This will include only the `Cabin` type and all types transitively referenced by it from the selection query.
+
+#### Flatten Naming Mode
+
+Use the `--flatten-naming` flag to flatten nested object structures into a single message with prefixed field names. This mode works with the selection query to flatten all root-level types selected in the query:
+
+```bash
+s2dm export protobuf --schema schema.graphql --selection-query query.graphql --output vehicle.proto --flatten-naming
+```
+
+You can optionally combine it with `--root-type` to flatten only a specific root type:
+
+```bash
+s2dm export protobuf --schema schema.graphql --selection-query query.graphql --output vehicle.proto --root-type Vehicle --flatten-naming
+```
+
+**Example transformation:**
+
+Given a GraphQL schema:
+
+```graphql
+type Vehicle {
+  adas: ADAS
+}
+
+type ADAS {
+  abs: ABS
+}
+
+type ABS {
+  isEngaged: Boolean
+}
+```
+
+Flatten mode produces:
+
+```protobuf
+syntax = "proto3";
+
+import "google/protobuf/descriptor.proto";
+import "buf/validate/validate.proto";
+
+extend google.protobuf.MessageOptions {
+  string source = 50001;
+}
+
+message Message {
+  bool Vehicle_adas_abs_isEngaged = 1;
+}
```

Comment:
Example is incomplete and not replicable because selection query is mandatory and not in the example.